scripts/build/builders/nrf.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import shlex
+
+from enum import Enum, auto
+
+from .builder import Builder
+
+
+class NrfApp(Enum):
+  LIGHT = auto()
+  LOCK = auto()
+  SHELL=auto()
+
+  def ExampleName(self):
+    if self == NrfApp.LIGHT:
+      return 'lighting-app'
+    elif self == NrfApp.LOCK:
+      return 'lock-app'
+    elif self == NrfApp.SHELL:
+      return 'shell'
+    else:
+      raise Exception('Unknown app type: %r' % self)
+
+  def AppNamePrefix(self):
+    if self == NrfApp.LIGHT:
+      return 'chip-nrf-lighting-example'
+    elif self == NrfApp.LOCK:
+      return 'chip-nrf-lock-example'
+    elif self == NrfApp.SHELL:
+      return 'chip-nrf-shell'
+    else:
+      raise Exception('Unknown app type: %r' % self)
+
+
+class NrfBoard(Enum):
+  NRF52840 = auto()
+  NRF5340 = auto()
+
+  def GnArgName(self):
+    if self == NrfBoard.NRF52840:
+      return 'nrf52840dk_nrf52840'
+    elif self == NrfBoard.NRF5340:
+      return 'nrf5340dk_nrf5340_cpuapp'
+    else:
+      raise Exception('Unknown board type: %r' % self)
+
+
+class NrfConnectBuilder(Builder):
+
+  def __init__(self,
+               root,
+               runner,
+               output_dir: str,
+               app: NrfApp = NrfApp.LIGHT,
+               board: NrfBoard = NrfBoard.NRF52840):
+    super(NrfConnectBuilder, self).__init__(root, runner, output_dir)
+    self.app = app
+    self.board = board
+
+  def generate(self):
+    if not os.path.exists(self.output_dir):
+        # NRF does a in-place update  of SDK tools
```

Comment:
Well, it does not automatically. It only calls:
```
COMMAND ${Python3_EXECUTABLE} scripts/setup/nrfconnect/update_ncs.py --check --quiet || (exit 0)
```
which prints a warning when your NCS version differs from the currently supported (and verified on CI).

Reply:
I had copied the update command from some of the build scripts, hence the comment. Since now I added the check, I  updated the comment as well.